marshal/tinyint/unmarshal_utils.go

```diff
@@ -0,0 +1,337 @@
+package tinyint
+
+import (
+	"errors"
+	"fmt"
+	"github.com/gocql/gocql/marshal"
+	"math/big"
+	"reflect"
+	"strconv"
+)
+
+var errUnmarshal = errors.Join(marshal.ErrorUnmarshal, errors.New("tinyint"))
+var errWrongDataLen = errors.Join(errUnmarshal, errors.New("the length of the data should be no more than 1"))
```

Comment:
I just realized that you may actually avoid extra error check if you do wrapping like this on `tinyint` package level.
So, can you please remove `marshal.ErrorUnMarshal` from here, and wrap errors in `gocql.Unmarhsal` level.
But keep `errUnmarshal` in.

```suggestion
```
```suggestion
var errUnmarshal = errors.New("failed to unmarshal tinyint")
var errWrongDataLen = errors.Join(errUnmarshal, errors.New("the length of the data should less then 2"))
```

Reply:
Done
```
var errWrongDataLen = errors.Join(errUnmarshal, errors.New("the length of the data should less or equal then 1"))
```